cli/lsp/documents.rs

```diff
@@ -290,28 +291,27 @@ impl Document {
   fn new(
     specifier: ModuleSpecifier,
     fs_version: String,
-    maybe_headers: Option<&HashMap<String, String>>,
-    content: Arc<str>,
+    maybe_headers: Option<HashMap<String, String>>,
+    text_info: SourceTextInfo,
     maybe_resolver: Option<&dyn deno_graph::source::Resolver>,
   ) -> Self {
     // we only ever do `Document::new` on on disk resources that are supposed to
     // be diagnosable, unlike `Document::open`, so it is safe to unconditionally
     // parse the module.
     let (maybe_module, maybe_parsed_source) = lsp_deno_graph_analyze(
       &specifier,
-      content.clone(),
-      maybe_headers,
+      text_info.text(),
+      maybe_headers.as_ref(),
       maybe_resolver,
     );
     let dependencies =
       Arc::new(DocumentDependencies::from_maybe_module(&maybe_module));
-    // todo(dsherret): retrieve this from the parsed source if it exists
```

Comment:
Not relevant anymore. We don't keep the parsed sources in memory.

Reply:
Wait, I'm wrong here. We do in maybe_parsed_source. Will look into this

Reply:
It should use less memory now because the parsed source and document will share the same SourceTextInfo.

Reply:
Very nice 🎉